python/ray/dag/compiled_dag_node.py

```diff
@@ -253,18 +256,26 @@ def _preprocess(self) -> None:
 
         # Find the input node to the DAG.
         for idx, task in self.idx_to_task.items():
-            if isinstance(task.dag_node, InputNode):
-                assert self.input_task_idx is None, "more than one InputNode found"
-                self.input_task_idx = idx
+            if isinstance(task.dag_node, InputNode) or isinstance(
+                task.dag_node, InputAttributeNode
+            ):
+                self.input_task_idxs.append(idx)
         # TODO: Support no-input DAGs (use an empty object to signal).
-        if self.input_task_idx is None:
+        if len(self.input_task_idxs) == 0:
             raise NotImplementedError(
                 "Compiled DAGs currently require exactly one InputNode"
             )
 
         # Find the (multi-)output node to the DAG.
         for idx, task in self.idx_to_task.items():
             if len(task.downstream_node_idxs) == 0:
+                # for multi-arg scenario, input args will go through
+                # InputAttributeNodes, InputNode will not connect to any node
```

Comment:
I didn't quite understand this comment. What does "go through" mean here? Maybe it would help to draw an example DAG?

Reply:
rewrote and added a diagram